test/python/test_reactor.py

```diff
@@ -785,6 +785,55 @@ def test_bad_kwarg(self):
         with self.assertRaises(TypeError):
             r1 = self.reactorClass(foobar=3.14)
 
+class TestMoleReactor(TestReactor):
+    reactorClass = ct.MoleReactor
+
+    def test_mole_reactor_surface_chem(self):
+        model = "ptcombust.yaml"
+        # initial conditions
+        T0 = 1500
+        P0 = ct.one_atm
+        equiv_ratio = 1
+        surf_area = 0.527
+        fuel = "CH4"
+        air = "O2:1.0, N2:3.773"
+        # reactor 1
+        gas1 = ct.Solution(model, transport_model=None)
+        gas1.TP = T0, P0
+        gas1.set_equivalence_ratio(equiv_ratio, fuel, air)
+        r1 = self.reactorClass(gas1)
+        # comparison reactor
+        gas2 = ct.Solution(model, transport_model=None)
+        gas2.TP = T0, P0
+        gas2.set_equivalence_ratio(equiv_ratio, fuel, air)
+        if "ConstPressure" in r1.type:
+            r2 = ct.ConstPressureReactor(gas2)
+        else:
+            r2 = ct.Reactor(gas2)
+        # surf 1
+        surf1 = ct.Interface(model, "Pt_surf", [gas1])
+        surf1.TP = T0, P0
+        surf1.coverages = {"PT(S)":1}
+        rsurf1 = ct.ReactorSurface(surf1, r1, A=surf_area)
+        # surf 2
+        surf2 = ct.Interface(model, "Pt_surf", [gas2])
+        surf2.TP = T0, P0
+        surf2.coverages = {"PT(S)":1}
+        rsurf2 = ct.ReactorSurface(surf2, r2, A=surf_area)
+        # reactor network setup
+        net1 = ct.ReactorNet([r1,])
+        net2 = ct.ReactorNet([r2,])
+        net1.rtol = net2.rtol = 1e-9
+        net1.atol = net2.atol = 1e-18
+        for i in np.arange(0.1, 1.1, 0.1):
+            net1.advance(i)
+            net2.advance(i)
```

Comment:
You should give some consideration to the time step and interval that's being tested here. For the problem at hand, the system has basically reached steady-state long before the first output time.